File: loom-compiler-old/src/main/java/loom/compiler/ast/statements/IfStmt.java
package loom.compiler.ast.statements;

import loom.compiler.ast.Expr;
import loom.compiler.ast.NodeVisitor;
import loom.compiler.ast.Stmt;
import loom.compiler.token.TokenSpan;

import java.util.List;

public class IfStmt implements Stmt {
	public final Expr condition;
	public final List<Stmt> thenBranch;
	public final List<Stmt> elseBranch; // null if no else branch
	public final TokenSpan span;

	public IfStmt(Expr condition, List<Stmt> thenBranch, List<Stmt> elseBranch, TokenSpan span) {
		this.condition = condition;
		this.thenBranch = thenBranch;
		this.elseBranch = elseBranch;
		this.span = span;
	}

	@Override
	public <T> T accept(NodeVisitor<T> visitor) {
		return visitor.visitIfStmt(this);
	}

	@Override
	public TokenSpan getSpan() {
		return span;
	}
} 

File: Frontend/Parser/AST/Statements/IfStmt.cs
// Copyright (c) 2026.The Cloth contributors.
// 
// IfStmt.cs is part of the Cloth Frontend.
// 
// Use, modification, and distribution of this file are governed by the
// license terms provided with the Cloth Compiler source distribution.

using FrontEnd.Parser.AST.Expressions;
using FrontEnd.Token;

namespace FrontEnd.Parser.AST.Statements;

public readonly record struct IfStmt(Expression Condition, Block ThenBranch, List<ElseIfBranch> ElseIfBranches, Block? ElseBranch, TokenSpan Span);